app/src/main/java/app/revanced/manager/ui/viewmodel/MainViewModel.kt

```diff
@@ -30,4 +41,70 @@ class MainViewModel(
             }
         }
     }
-}
\ No newline at end of file
+
+    fun launchLegacyActivity(): Boolean {
+        return try {
+            val intent = Intent()
+            intent.setClassName(
+                "app.revanced.manager.flutter",
+                "app.revanced.manager.flutter.ExportSettingsActivity"
+            )
+            launcher.launch(intent)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    fun applyLegacySettings(data: String) = viewModelScope.launch {
+        val json = Json { ignoreUnknownKeys = true }
+        val settings = json.decodeFromString<LegacySettings>(data)
+
+        settings.themeMode?.let { theme ->
+            val themeMap = mapOf(
+                0 to Theme.SYSTEM,
+                1 to Theme.LIGHT,
+                2 to Theme.DARK
+            )
+            prefs.theme.update(themeMap[theme]!!)
+        }
+        settings.useDynamicTheme?.let { dynamicColor ->
+            prefs.dynamicColor.update(dynamicColor == 1)
+        }
+        settings.apiUrl?.let { api ->
+            prefs.api.update(api.removeSuffix("/"))
+        }
+        settings.experimentalPatchesEnabled?.let { allowExperimental ->
+            prefs.allowExperimental.update(allowExperimental == 1)
+        }
+        settings.patchesAutoUpdate?.let { autoUpdate ->
+            with(patchBundleRepository) {
+                sources
+                    .first()
+                    .find { it.uid == 0 }
+                    ?.asRemoteOrNull
+                    ?.setAutoUpdate(autoUpdate == 1)
+
+                updateCheck()
+            }
+        }
+        settings.patchesChangeEnabled?.let {
+            // TODO: Implement setting
+        }
+        settings.showPatchesChangeWarning?.let {
+            // TODO: Implement setting
+        }
+        settings.keystore?.let { keystore ->
+            prefs.keystoreCommonName.update("ReVanced")
+            prefs.keystorePass.update(settings.keystorePassword)
+            val keystorePath = app.getDir("signing", Context.MODE_PRIVATE)
+                .resolve("manager.keystore").toPath()
+            val keystoreBytes = Base64.decode(keystore, Base64.DEFAULT)
+            keystorePath.toFile().writeBytes(keystoreBytes)
```

Comment:
Try to use `KeystoreManager` to import it instead.
Make sure to handle exceptions.